plugins/kinet/KiNetNode.cpp

```diff
@@ -177,9 +183,9 @@ void KiNetNode::SocketReady() {
  * Fill in the header for a packet
  */
 void KiNetNode::PopulatePacketHeader(uint16_t msg_type) {
-  uint32_t sequence_number = 0;  // everything seems to set this to 0.
   m_output_stream << KINET_MAGIC_NUMBER << KINET_VERSION_ONE;
-  m_output_stream << msg_type << sequence_number;
+  m_output_stream << msg_type << HostToBigEndian(m_packet_number);
+  m_packet_number++;
```

Comment:
> But it's one global counter between all the different packet types, not one per packet type?

Yes. The counter appears to increment for each packet sent by QuickPlay Pro 2 no matter the packet type or device. I just did some more testing, and it seems that this sequence number is also used by end devices when responding to a packet. It seems that the end device responds with the same sequence number to match a response to a request.

>So just to check, this should also increment with a DMXOUT packet (for example)? 

Yes, I believe so. I don't know how to make QuickPlay Pro 2 produce a DMXOUT packet, so I cannot test that specifically. But every single packet put out by QuickPlay Pro 2 seems to have this counter in it.

Reply:
> Yes. The counter appears to increment for each packet sent by QuickPlay Pro 2 no matter the packet type or device. I just did some more testing, and it seems that this sequence number is also used by end devices when responding to a packet.

Ah, so actually the sequence number will help us to make sense of the endian stuff, can you do a capture for when the sequence number rolls over from 0x00ff to 0x0100 and that should confirm the endianness of that counter too.

> It seems that the end device responds with the same sequence number to match a response to a request.

Ah, that was going to be my next question! So they reply with the same sequence number value.

> Yes, I believe so. I don't know how to make QuickPlay Pro 2 produce a DMXOUT packet, so I cannot test that specifically. But every single packet put out by QuickPlay Pro 2 (not just PORTOUT) seems to have this counter in it.

Ah, I thought I found a way to do that, but maybe I'm getting mixed up. The other packets confirm it though!

Reply:
> Ah, so actually the sequence number will help us to make sense of the endian stuff, can you do a capture for when the sequence number rolls over from 0x00ff to 0x0100 and that should confirm the endianness of that counter too.

I compared side by side what my commits spit out and what QuickPlay Pro 2 spits out, and they currently match (I tested rolling the first 3 bytes). So that should tell you what it should be.

The order I am seeing in packet capture is:
`0x0801` `0x01000000` `0xffffffff`
`0x0801` `0x02000000` `0xffffffff`
...
`0x0801` `0xff000000` `0xffffffff`
`0x0801` `0x00010000` `0xffffffff`
`0x0801` `0x01010000` `0xffffffff`